WebAPI/Controllers/ReservationsController.cs

```diff
@@ -42,4 +44,28 @@ public async Task<ActionResult<ReservationDto>> GetReservationAsync(int id)
         }
         return NotFound();
     }
+
+    /// <summary>
+    /// Creates the reservation entity based on specified parameter
+    /// </summary>
+    /// <param name="reservationDto">Object that contains parameters from which new entity is being created</param>
+    /// <returns>Id of the created object</returns>
+    [HttpPost]
+    [SwaggerResponse((int)HttpStatusCode.Created)]
+    [SwaggerResponse((int)HttpStatusCode.BadRequest)]
+    [SwaggerResponse((int)HttpStatusCode.UnprocessableEntity, Description = "Validation Error")]
+    [SwaggerResponse((int)HttpStatusCode.InternalServerError)]
+    public async Task<ActionResult> CreateReservationAsync([FromBody] CreateReservationDto reservationDto)
+    {
+        var createReservationCommand = new CreateReservationCommand(reservationDto);
+
+        var id= await this._mediator.Send(createReservationCommand, new CancellationToken());
+
+        if (id > 0)
+        {
+            return Created("reservations/{id}", id);
+        }
+
+        return BadRequest("Something went wrong");
```

Comment:
BadRequest je na nivou 400 - klijentska greška. Ovde to ne bi trebalo da bude slučaj (validacija za komandu bi trebalo da spreči slučaj gde klijent pogreši).
To znači da se ovde nešto na serverskoj strani loše desilo, išao bi 500 Internal Server Error - reservation not created successfully ili tako nešto